### PR TITLE
Fix potential panic in create_http_context

### PR DESCRIPTION
**Pull request by junr03**
Signed-off-by: José Ulises Niño Rivera <junr03@users.noreply.github.com>

### CODE REVIEW

File: arch/src/filter_context.rs [hunk 1]
```diff
@@ -282,7 +282,10 @@ impl RootContext for FilterContext {
         // No StreamContext can be created until the Embedding Store is fully initialized.
         let embedding_store = match self.mode {
             GatewayMode::Llm => None,
-            GatewayMode::Prompt => Some(Rc::clone(self.embeddings_store.as_ref().unwrap())),
```
Comment (junr03) [inline arch/src/filter_context.rs:285]: this unwrap can panic if a stream is being created before the embeddings store has finished initializing

File: arch/src/filter_context.rs [hunk 2]
```diff
@@ -282,7 +282,10 @@ impl RootContext for FilterContext {
...
+            GatewayMode::Prompt => match self.embeddings_store.as_ref() {
+                None => return None,
```
Comment (adilhafeez) [inline arch/src/filter_context.rs:286]:
```suggestion
                None => None,
```

**Review by adilhafeez — APPROVED** [review]

File: arch/src/filter_context.rs (hunk 2 above)
Comment (junr03) [reply arch/src/filter_context.rs:286]: I did intend to `return None`. The whole point here is that a StreamContext should not be created if the Embedding Store is not present

Comment (adilhafeez) [reply arch/src/filter_context.rs:286]: ah I see it now! make sense.

**Comment by adilhafeez** [comment]
fixed in https://github.com/katanemo/arch/pull/194

**Pull request closed without merging**